compiler/ir/backend.jvm/lower/src/org/jetbrains/kotlin/backend/jvm/JvmLower.kt

```diff
@@ -142,7 +142,13 @@ internal val localDeclarationsPhase = makeIrFilePhase(
                 private fun scopedVisibility(inInlineFunctionScope: Boolean): DescriptorVisibility =
                     if (inInlineFunctionScope) DescriptorVisibilities.PUBLIC else JavaDescriptorVisibilities.PACKAGE_VISIBILITY
             },
-            forceFieldsForInlineCaptures = true
+            forceFieldsForInlineCaptures = true,
+            postLocalDeclarationLoweringCallback = { (localFunctions, newParameterToOld, oldParameterToNew, newParameterToCaptured) ->
```

Comment:
@udalov I realize, preparing the PR, that the current version of this PR is order dependent: only the last invocation of LocalDeclarationsLowering's intermediate data structures will be recorded.

I think I just need to lift this to a map from containing IrFile/IrModule->Intermediate structure, and can quickly do so -- but do you agree with the overall approach of "state passing" via the context?

Reply:
It seems fine at first glance, but I would prefer if this was only done in case the fragment evaluation is enabled, and skipped if this is the normal compilation mode.

Reply:
I pushed a revision, and think I have a design that avoids work in the normal case to the greatest extent possible.